common/djangoapps/config_models/admin.py

```diff
@@ -78,3 +88,108 @@ def revert(self, request, queryset):
+        return [self.parameter_name]
+
+
+class KeyedConfigurationModelAdmin(ConfigurationModelAdmin):
+    """
+    :class:`~django.contrib.admin.ModelAdmin` for :class:`.ConfigurationModel` subclasses that
+    use extra keys (i.e. they have KEY_FIELDS set).
+    """
+    date_hierarchy = None
+    list_filter = (ShowHistoryFilter, )
+
+    def queryset(self, request):
+        """
+        Annote the queryset with an 'is_active' property that's true iff that row is the most
+        recently added row for that particular set of KEY_FIELDS values.
+        Filter the queryset to show only is_active rows by default.
+        """
+        queryset = super(KeyedConfigurationModelAdmin, self).queryset(request)
+        key_fields_escaped = [connection.ops.quote_name(name) for name in self.model.KEY_FIELDS]
+        # The following custom SQL is like a "SELECT DISTINCT ON {key_fields} ..." query, which
+        # is only supported on PostgreSQL.
+        subquery = """
+            SELECT id FROM
+                (SELECT id, {key_fields} FROM {table_name} AS ordered_table ORDER BY change_date DESC) AS grouped_table
+            GROUP BY {key_fields}
+        """.format(key_fields=', '.join(key_fields_escaped), table_name=self.model._meta.db_table)
+        if request.GET.get(ShowHistoryFilter.parameter_name) == '1':
+            # Show all rows but annotate them with an 'is_active' boolean
+            queryset = queryset.extra(
+                select={'is_active': "id IN ({subquery})".format(subquery=subquery)}
+            )
+        else:
+            # Show only the most recent row for each key:
+            queryset = queryset.extra(
+                select={'is_active': True},
+                where=["id IN ({subquery})".format(subquery=subquery)]
+            )
+        return queryset.order_by('-change_date')
+
+    def get_list_display(self, request):
+        """ Add a link to each row for creating a new row using the chosen row as a template """
+        return self.model._meta.get_all_field_names() + ['edit_link']
+
+    def add_view(self, request, form_url='', extra_context=None):
+        # Prepopulate new configuration entries with the value of the current config
+        if 'source' in request.GET:
+            get = request.GET.copy()
+            source_id = int(get.pop('source')[0])
+            source = self.model.objects.get(pk=source_id)
+            get.update(models.model_to_dict(source))
+            request.GET = get
+        # Call our grandparent's add_view, skipping the parent code. pylint: disable=bad-super-call
+        return super(ConfigurationModelAdmin, self).add_view(request, form_url, extra_context)
+
+    def edit_link(self, inst):
+        """ Edit link for the change view """
+        if not inst.is_active:
+            return u'--'
+        update_url = reverse('admin:{}_{}_add'.format(self.model._meta.app_label, self.model._meta.module_name))
+        update_url += "?source={}".format(inst.pk)
```

Comment:
Nit: this should generally be done with `urllib.urlencode()` to ensure proper escaping.  (Not really relevant in this case, since the primary key is numeric.)